metadata/src/main/java/org/apache/kafka/image/node/ProducerIdsImageNode.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.image.node;
+
+import org.apache.kafka.image.ProducerIdsImage;
+
+import java.util.Collection;
+import java.util.Collections;
+
+
+public class ProducerIdsImageNode implements MetadataNode {
+    /**
+     * The name of this node.
+     */
+    public static final String NAME = "producerIds";
+
+    /**
+     * The producer IDs image.
+     */
+    private final ProducerIdsImage image;
+
+    public ProducerIdsImageNode(ProducerIdsImage image) {
+        this.image = image;
+    }
+
+    @Override
+    public Collection<String> childNames() {
+        return Collections.singletonList("nextProducerId");
+    }
+
+    @Override
+    public MetadataNode child(String name) {
+        if (name.equals("nextProducerId")) {
+            return new MetadataLeafNode(image.highestSeenProducerId() + "");
```

Comment:
Looking at the code, it actually is the next producer ID and *not* the "highest seen", except in the special case where we haven't issued any producer IDs, where it's -1. That special case is a bit silly... we could have just had it be 0 there.

In the future we could just use 0 instead of -1 as the start value to avoid the special case. But in the interest of keeping this change small I'll rename the accessor to `nextProducerId` and add a JavaDoc comment about the special case. The accessor is just used for debugging and so on in any case.